plugins/hosts/darwin/cap/fs_iso.rb

```diff
@@ -0,0 +1,62 @@
+require 'pathname'
+require "vagrant/util/subprocess"
+require "vagrant/util/map_command_options"
+require "vagrant/util/directory_changed"
+require "vagrant/util/ensure_output_file"
+
+module VagrantPlugins
+  module HostDarwin
+    module Cap
+      class FsISO
+        extend Vagrant::Util::EnsureOutputFile
+
+        @@logger = Log4r::Logger.new("vagrant::host::darwin::fs_iso")
+
+        BUILD_ISO_CMD = "hdiutil".freeze
+
+        # Check that the host has the ability to generate ISOs
+        #
+        # @param [Vagrant::Environment] env
+        # @return [Boolean]
+        def self.isofs_available(env)
+          !!Vagrant::Util::Which.which("hdiutil")
+        end
+
+        # Generate an ISO file of the given source directory
+        #
+        # @param [Vagrant::Environment] env
+        # @param [String] source_directory Contents of ISO
+        # @param [String, nil] file_destination Location to store ISO
+        # @param [Map] extra arguments to pass to the iso building command
+        # @return [Pathname] ISO location
+        # @note If file_destination exists, source_directory will be checked
+        #       for recent modifications and a new ISO will be generated if requried.
+        def self.create_iso(env, source_directory, file_destination=nil, extra_opts={})
+          file_destination = ensure_output_file(ext=".iso", file_destination=file_destination)
```

Comment:
I think there's an advantage to keeping this code in the cap, at least until we need a similar check some place else. If there's a concern that the `file_destination` doesn't end in `.iso`, we could also check `File.extname(file_destination)` here.

```suggestion
          if file_destination.nil?
            tmpfile = Tempfile.new(["vagrant", ".iso"])
            file_destination = Pathname.new(tmpfile.to_path)
            tmpfile.delete
          else
            file_destination = Pathname.new(file_destination.to_s)
            FileUtils.mkdir_p(File.dirname(file_destination))
          end
```

Reply:
My reasoning for pulling this out into a util was that a similar check will be needed for the linux and windows iso caps. What do you think about extracting this bit of code into this util (to be used by the darwin, linux and windows caps) and dropping the bit about ensuring the file ends in some extension?

Reply:
The way I read it, the first check `if file_destination.nil?` is providing a default value for the `file_destination` argument to the method. The second clause is the one that could be better candidate to be extracted out, conceptually ("make sure my target destination exists"), but since that's not very much code I don't see a lot of value in making that a helper method.